src/python_framework_v01/nhd_network_utilities_v01.py

```diff
@@ -767,3 +773,19 @@ def get_reservoir_restart_from_wrf_hydro(
     resdf = resdf.set_index(["Waterbody"])
     init_waterbody_states = resdf
     return init_waterbody_states
+
```

Comment:
I line 747 above, the function should consider several cases:
1. A one-to-one match between the number of reservoirs described in there restart file and those in the LAKEPARM input file. If this is the case, no crosswalk is needed, the files are connected one-to-one in the order as they are.
1. A one-to-one match as above, but with a crosswalk file supplied. files are connected per the crosswalk.
1. More reservoirs in the restart file than in the LAKEPARM -- this should be flagged with a warning in debuglevel 1!! It could be valid if we are hotstarting a subset model with a restart file from a larger domain. THIS CASE REQUIRES AN ID MAPPING FILE and should throw and error if it is not provided.  
1. More reservoirs in the LAKEPARM than in the restart file. This should also be flagged with a warning in debuglevel 1  -- it is probably similar to the above, with a LAKEPARM file from a larger domain.  There are at least two ways to handle the ordering here. 1) use an ID mapping file. 2) the order in the LAKEPARM file can be assumed to be the order for the ID mapping. 

__For both of these last two cases, if the number of reservoirs referred to in the route-link file is different than the number referred to in the smaller of the number in either the restart file or the LAKEPARM file, then we cannot continue__

Reply:
So this leads to a function signature like this:
Func(
Restart file _required_
List_of_reservoirs_in_route_link _file, ordered how you want to match _required_
)

Do the ordering in a separate function first. 
If the lengths of these two don't match, __STOP__

Reply:
Looking for clarification on this one as well. The LAKEPARM file currently has 5459 rows × 15 columns and the route-link file has 504 rows × 1 columns of lakes within it. These can't match as they are what we are using to order the lake ids in the first place? The LAKEPARM has the correct order and we use the route-link to pull the IDs, correct?